packages/mobile/src/components/core/UserGeneratedText.tsx

```diff
@@ -306,10 +320,19 @@ export const UserGeneratedText = (props: UserGeneratedTextProps) => {
             text={squashNewLines(children) as string}
             matchers={[
               // Handle matcher e.g. @handle
-              {
-                pattern: /@[a-zA-Z0-9_.]+/,
-                renderLink: renderHandleLink
-              },
+              ...(mentions
+                ? [
+                    {
+                      pattern: mentionRegex,
+                      renderLink: renderHandleLink
+                    }
+                  ]
+                : [
+                    {
+                      pattern: handleRegex,
+                      renderLink: renderHandleLink
+                    }
+                  ]),
```

Comment:
Wait I'm confused; why do we have both regexes in here?

Reply:
one is for comment when we only want to highlight the handles that have explicitly been mentioned, the other is for DMs and other instances where we want to highlight the handles always